lib/youtube/queryAnalyticsReports.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * YouTube Analytics Client Utilities
+ * 
+ * Provides reusable functions for querying YouTube Analytics API
+ * with proper authentication and error handling.
+ */
+
+import { createYouTubeAnalyticsClient } from "./youtube-analytics-oauth-client";
+import { createYouTubeAPIClient } from "./oauth-client";
+import { AnalyticsReportsResult } from "@/types/youtube";
+import { TokenValidation } from "@/types/youtube";
+
+/**
+ * Query YouTube Analytics reports for specified metrics and date range
+ * @param tokenValidation - Validated YouTube tokens
+ * @param startDate - Start date in YYYY-MM-DD format
+ * @param endDate - End date in YYYY-MM-DD format
+ * @param metrics - Analytics metrics to query (e.g., "estimatedRevenue", "views")
+ * @returns Analytics data with daily breakdown and totals
+ */
+export async function queryAnalyticsReports(
+  tokenValidation: TokenValidation,
+  startDate: string,
+  endDate: string,
+  metrics: string
+): Promise<AnalyticsReportsResult> {
+  // Get user's channel ID first
+  const youtube = createYouTubeAPIClient(
+    tokenValidation.tokens!.access_token, 
+    tokenValidation.tokens!.refresh_token ?? undefined
+  );
+  
+  const channelResponse = await youtube.channels.list({
+    part: ["id"],
+    mine: true,
+  });
+
+  if (!channelResponse.data.items || channelResponse.data.items.length === 0) {
+    throw new Error("No YouTube channel found for this account. Please ensure you have a YouTube channel.");
+  }
+
+  const channelId = channelResponse.data.items[0].id;
+  if (!channelId) {
+    throw new Error("Unable to retrieve channel ID. Please ensure your YouTube account is properly set up.");
+  }
+
+  // Create YouTube Analytics API client
+  const ytAnalytics = createYouTubeAnalyticsClient(
+    tokenValidation.tokens!.access_token,
+    tokenValidation.tokens!.refresh_token ?? undefined
+  );
+
+  // Query analytics reports for the specified date range
+  const response = await ytAnalytics.reports.query({
+    ids: `channel==${channelId}`,
+    startDate: startDate,
+    endDate: endDate,
+    metrics: metrics,
```

Comment:
same thing with less code right?
```suggestion
    startDate,
    endDate,
    metrics,
```